substrate/client/api/src/client.rs

```diff
@@ -164,6 +164,12 @@ pub trait BlockBackend<Block: BlockT> {
 	fn requires_full_sync(&self) -> bool;
 }
 
+/// Tells whether the current client configuration requires proof recording.
+pub trait EnableProofRecording {
+	/// Tells whether the current client configuration requires proof recording.
+	fn enable_proof_recording(&self) -> bool;
+}
```

Comment:
The trait name could be more generic, to possibly allow other uses in the future.

```suggestion
/// Tells whether the current client configuration requires proof recording.
pub trait ProofRecording {
	/// Tells whether the current client configuration requires proof recording.
	fn proof_recording_enabled(&self) -> bool;
}
```